cfme/automate/import_export.py

```diff
@@ -1,70 +1,112 @@
+from cfme.utils.appliance.implementations.ui import navigator
 
 
-class GitImportSelectorView(AutomateImportExportBaseView):
-    type = BootstrapSelect('branch_or_tag')
-    branch = BootstrapSelect(locator='.//div[contains(@class, "bootstrap-select git-branches")]')
-    tag = BootstrapSelect(locator='.//div[contains(@class, "bootstrap-select git-tags")]')
+class AutomateImportExportBaseView(BaseLoggedInPage):
+    flash = FlashMessages("div.import-flash-message")
+    title = Text('.//div[@id="main-content"]//h1')
 
-    submit = Button('Submit')
-    cancel = Button('Cancel')
+    @property
+    def in_import_export(self):
+        return (
+            self.logged_in_as_current_user
+            and self.navigation.currently_selected == ["Automation", "Automate", "Import / Export"]
+            and self.title.text == "Import / Export"
+        )
 
     @property
     def is_displayed(self):
-        return self.in_import_export and self.type.is_displayed
+        return self.in_import_export
 
 
-class AutomateGitRepository(Navigatable):
-    """Represents an Automate git repository. This entity is not represented in UI as it is, but
-    only in database. But by representing it it makes the code changes for domain much simpler.
+class AutomateImportExportView(AutomateImportExportBaseView):
+    @View.nested
+    class import_file(View):  # noqa
+        file = FileInput(id="upload_file")
+        upload = Button(id="upload-datastore-import")
 
-    """
-    def __init__(self, url=None, username=None, password=None, verify_ssl=None, appliance=None):
-        Navigatable.__init__(self, appliance=appliance)
-        self.url = url
-        self.username = username
-        self.password = password
-        self.verify_ssl = verify_ssl
-        self.domain = None
+    @View.nested
+    class import_git(View):  # noqa
+        ROOT = './/form[@id="retrieve-git-datastore-form"]'
 
-    @classmethod
-    def from_db(cls, db_id, appliance=None):
-        git_repositories = appliance.db.client['git_repositories']
-        try:
-            url, verify_ssl = appliance.db.client.session\
-                .query(git_repositories.url, git_repositories.verify_ssl)\
-                .filter(git_repositories.id == db_id)\
-                .first()
-            return cls(url=url, verify_ssl=verify_ssl > 0, appliance=appliance)
-        except ValueError:
-            raise ValueError('No such repository in the database')
+        url = Input(name="git_url")
+        username = Input(name="git_username")
+        password = Input(name="git_password")
+        verify_ssl = Checkbox(name="git_verify_ssl")
+        submit = Button(id="git-url-import")
+
+    export_all = Button(title="Export all classes and instances")
+    reset_all = Button(title="Reset all components in the following domains: RedHat, ManageIQ")
+    reset_title = Text(
+        ".//div[contains(@class, 'import-or-export')]/h3[contains(text(), 'Reset all')]"
+    )
```

Comment:
@kedark3 one important thrum rule which i learn about view. Always write view as per `UI widget Location`. `Up to down and left to right`.
where it will help?
`view.fill({...})` will fill widgets as per widget declaration in view.

Reply:
Got it. I wasn't aware of that order. Its good to know :+1:  Thanks.

Reply:
Literally went through this exact thought process as I was reading, nice work guys!